app/modules/tmr.c

```diff
@@ -125,17 +125,24 @@ static int tmr_register(lua_State* L) {
 	return 0;
 }
 
-// Lua: t:start()
+// Lua: t:start( [restart] )
 static int tmr_start(lua_State* L){
 	tmr_t *tmr = (tmr_t *) luaL_checkudata(L, 1, "tmr.timer");
-  int idle = tmr->mode & TIMER_IDLE_FLAG;
+	lua_settop(L, 2);
+	luaL_argcheck(L, lua_isboolean(L, 2) || lua_isnil(L, 2), 2, "boolean expected");
+	int restart = lua_isboolean(L, 2) ? lua_toboolean(L, 2) : 0;
```

Comment:
```suggestion
	int restart = lua_isboolean(L, 2);
```
You already verified it's a boolean in the line above.

Reply:
But it can still be `nil` if default parameter is used. I have not checked that `lua_toboolean(L, 2)` returns 0 if parameter is `nil`. So 👍